### PR TITLE
GRW-631 / Feature / CPAS Button click tracking

### PR DESCRIPTION
**Pull request by amandaflod**
## What?

Add tracking to Footer buttons and back buttons in CPAS checkout. 

## Why?

**Ticket(s): [GRW-631]**

[GRW-631]: https://hedvig.atlassian.net/browse/GRW-631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/components/BackButton/BackButton.stories.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default {
 
 const Template: Story = () => (
   <MemoryRouter initialEntries={['/se-en/new-member/checkout/details']}>
-    <BackButton />
+    <BackButton onClick={() => console.log('click')} />
```
Comment (dendoesit) [inline src/client/components/BackButton/BackButton.stories.tsx:19]: we should remove the console log here.

File: src/client/components/BackButton/BackButton.tsx [hunk 2]
```diff
@@ -16,15 +16,16 @@ const Button = styled(UnstyledButton)`
   }
 `
 
-export const BackButton = () => {
+export const BackButton = ({ onClick }: { onClick: () => void }) => {
```
Comment (simonauner) [inline src/client/components/BackButton/BackButton.tsx:19]: Should perhaps be moved to a `BackButtonProps`? 

Perhaps this button is only ever used in `CheckoutPageWrapper`? If not, should `onClick` be optional?

Comment (amandaflod) [reply src/client/components/BackButton/BackButton.tsx:19]: Ok, will do that! :D I will make this PR into a draft again now and await Marian's hook to be merged, so there will be some more changes to this soon!

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 3]
```diff
@@ -47,6 +47,7 @@ export const CheckoutDetails = () => {
   }
 
   const handleOnClick = () => {
+    console.log('hello')
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx]: Remove ;)

File: src/client/utils/tracking/trackOfferEvent.ts [hunk 4]
```diff
@@ -118,6 +119,7 @@ export const useTrackOfferEvent = () => {
       { eventName, options = {} }: EventParameters,
       selectedBundleVariant: QuoteBundleVariant,
     ) => {
+      console.log('track', eventName, options)
```
Comment (robinandeer) [inline src/client/utils/tracking/trackOfferEvent.ts]: Remove ;)

File: src/client/components/BackButton/BackButton.tsx [hunk 5]
```diff
@@ -16,15 +20,19 @@ const Button = styled(UnstyledButton)`
...
+export const BackButton = ({ onClick }: Props) => {
   const history = useHistory()
 
-  const goBack = () => {
+  const handleClick = () => {
     history.goBack()
+
+    if (onClick) {
+      onClick()
+    }
```
Comment (robinandeer) [inline src/client/components/BackButton/BackButton.tsx:31]:
```suggestion
    onClick?.()
```

**Review by dendoesit — DISMISSED** [review]
looks good. we can fix the checkout details separately.

**Review by robinandeer — DISMISSED** [review]
Looks good to me I think :)

**Review by dendoesit — APPROVED** [review]
⭐

**Pull request merged**